File: backend/ErpPortfolio.Api/Models/Devolucao.cs
// =====================================================================================
// Arquivo....: Devolucao.cs
// Versão.....: 1.0.0
// Data.......: 24/09/2026
// Descrição..: Devolução (parcial ou total) de um pedido de venda confirmado (SPEC.md
//              etapa 14). Guarda os totais já decididos na hora: quanto abateu das parcelas
//              pendentes e quanto virou reembolso ao cliente (valor_total = abatido +
//              reembolso). É definitiva: não se edita nem se exclui.
// -------------------------------------------------------------------------------------
// Banco......: PostgreSQL - erp_portfolio_db
// Tabelas....: public.devolucoes (FK pedido_id -> public.pedidos)
// Fontes.....: Mapeada em ErpPortfolioDbContext.Devolucoes. Criada pelo DevolucaoService.
// -------------------------------------------------------------------------------------
// Histórico de alterações:
//   1.0.0 - 24/09/2026 - Criação do arquivo.
// =====================================================================================

namespace ErpPortfolio.Api.Models;

public class Devolucao
{
    /// <summary>Também é o número da devolução.</summary>
    public int Id { get; set; }

    public int PedidoId { get; set; }

    public Pedido? Pedido { get; set; }

    /// <summary>Data/hora (UTC) do registro.</summary>
    public DateTime DataDevolucao { get; set; }

    public string? Motivo { get; set; }

    /// <summary>Valor dos itens devolvidos, com os descontos do pedido (DV3).</summary>
    public decimal ValorTotal { get; set; }

    /// <summary>Parte descontada das parcelas pendentes (DV4).</summary>
    public decimal ValorAbatido { get; set; }

    /// <summary>Parte que o cliente já tinha pago e volta como conta a pagar (DV5).</summary>
    public decimal ValorReembolso { get; set; }

    public List<DevolucaoItem> Itens { get; set; } = [];
}

File: frontend/erp-portfolio-web/src/types/devolucao.ts
/**
 * =====================================================================
 * Arquivo....: devolucao.ts
 * Versão.....: 1.0.0
 * Data.......: 24/09/2026
 * Descrição..: Tipos da devolução de venda (etapa 14), espelhando DevolucaoCriacaoDto,
 *              DevolucaoRespostaDto e DevolucoesResumoDto da API.
 * ---------------------------------------------------------------------
 * Histórico de alterações:
 *   1.0.0 - 24/09/2026 - Criação do arquivo.
 * =====================================================================
 */

export interface DevolucaoItemEntrada {
  /** Id do item do pedido (não do produto). */
  pedidoItemId: number
  quantidade: number
  /** false = perda (ex.: defeito): não volta ao estoque. */
  voltaEstoque: boolean
}

export interface DevolucaoEntrada {
  itens: DevolucaoItemEntrada[]
  motivo: string | null
  /** AAAA-MM-DD; só usado se houver reembolso (padrão: hoje). */
  vencimentoReembolso: string | null
}

export interface DevolucaoItem {
  id: number
  pedidoItemId: number
  produtoId: number
  produtoNome: string
  unidade: string
  quantidade: number
  valor: number
  voltaEstoque: boolean
}

export interface Devolucao {
  /** Também é o número da devolução. */
  id: number
  pedidoId: number
  /** Data/hora ISO 8601 em UTC. */
  dataDevolucao: string
  motivo: string | null
  valorTotal: number
  /** Parte descontada das parcelas pendentes. */
  valorAbatido: number
  /** Parte que virou conta a pagar de reembolso ao cliente. */
  valorReembolso: number
  parcelaPagarId: number | null
  /** Estorno de comissão (negativo) ou 0. */
  estornoComissao: number
  itens: DevolucaoItem[]
}

/** Card "Devoluções do mês" do Dashboard. */
export interface DevolucoesResumo {
  valorTotal: number
  quantidade: number
}
